lib/mongoid/interceptable.rb

```diff
@@ -152,6 +152,30 @@ def _mongoid_run_child_callbacks(kind, children: nil, &block)
       end
     end
 
+    # This is used to store callbacks to be executed later. A good use case for
+    # this is delaying the after_find and after_initialize callbacks until the
+    # associations are set on the document. This can also be used to delay
+    # applying the defaults on a document.
+    #
+    # @return [ Array<Symbol> ] an array of symbols that represent the pending callbacks.
+    def pending_callbacks
+      @pending_callbacks ||= []
+    end
+
+    # Run the pending callbacks. If the callback is :default, we will apply
+    # the defaults for this document. Otherwise, the callback is passed to the
+    # run_callbacks function.
+    def run_pending_callbacks
+      pending_callbacks.each do |cb|
+        if cb == :default
```

Comment:
Can `:default` be renamed to `:apply_defaults`? That would make the callers easier to read.